engine/src/flutter/testing/dart/fragment_shader_test.dart

```diff
@@ -311,6 +314,65 @@ void main() async {
     blueGreenImage.dispose();
   });
 
+  test('FragmentShader renders sampler with filter quality default', () async {
+    final FragmentProgram program = await FragmentProgram.fromAsset('texture.frag.iplr');
+    final Image blueGreenImage = await _createBlueGreenImage();
+    final FragmentShader shader = program.fragmentShader()
+      ..setImageSampler(0, blueGreenImage)
+      ..setFloat(0, 100)
+      ..setFloat(1, 100);
+
+    final Image shaderImage = await _imageFromShader(shader: shader, imageDimension: 100);
+
+    // Default filter quality uses FilterQuality.none.
+    await comparer.addGoldenImage(shaderImage, 'render_sampler_none.png');
+    shader.dispose();
+    blueGreenImage.dispose();
+  });
+
+  test('FragmentShader renders sampler with filter quality low', () async {
+    final FragmentProgram program = await FragmentProgram.fromAsset('texture.frag.iplr');
+    final Image blueGreenImage = await _createBlueGreenImage();
+    final FragmentShader shader = program.fragmentShader()
+      ..setImageSampler(0, blueGreenImage, filterQuality: FilterQuality.low)
+      ..setFloat(0, 100)
+      ..setFloat(1, 100);
+
+    final Image shaderImage = await _imageFromShader(shader: shader, imageDimension: 100);
+
+    await comparer.addGoldenImage(shaderImage, 'render_sampler_low.png');
+    shader.dispose();
+    blueGreenImage.dispose();
```

Comment:
nit: These tests share all the same code except the filter quality.  A helper function could be factored out.